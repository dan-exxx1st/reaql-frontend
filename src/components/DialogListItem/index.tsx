import React, { FC } from 'react';
import StyledDialogListItem from './style';

const DialogListItem: FC = () => {
	return <StyledDialogListItem>Dialog list item</StyledDialogListItem>;
};

export default DialogListItem;
