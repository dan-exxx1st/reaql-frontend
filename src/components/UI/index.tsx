export { default as Avatar } from './Avatar';
export { default as Button } from './Button';
export { default as Circle } from './Circle';
export { default as SkeletonLoader } from './SkeletonLoader';
export { default as TextField } from './TextField';
export { default as Typography } from './Typography';
export { default as Wrapper } from './Wrapper';
export { default as CheckMark } from './CheckMark';
export { default as Icon } from './Icon';
export { default as CheckBox } from './CheckBox';
export { default as SystemMessage } from './SystemMessage';
export { default as Spinner } from './Spinner';
