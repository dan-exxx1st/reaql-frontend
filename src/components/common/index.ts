export { default as DialogHeader } from './DialogHeader';
export { default as ContactItem } from './ContactsItem';
export { default as UserHeader } from './UserHeader';
export { default as DialogTextField } from './DialogTextField';
export { default as MessageItem } from './MessageItem';
export { default as DialogHeaderBackdrop } from './DialogHeaderBackdrop';
export { default as MessageSearch } from './MessageSearch';
