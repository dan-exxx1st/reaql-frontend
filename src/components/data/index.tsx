export { default as MessageListWithData } from './MessageListWithData';
export { default as DialogHeaderWithData } from './DialogHeaderWithData';
