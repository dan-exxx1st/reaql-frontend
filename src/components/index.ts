export { default as DialogListItem } from './common/DialogListItem';
export { default as EmojiPicker } from './common/EmojiPicker';
