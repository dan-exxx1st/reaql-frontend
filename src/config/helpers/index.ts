import { useState, useEffect } from 'react';

export function useLoaded({ src }: { src?: HTMLImageElement['src'] }) {
	const [loaded, setLoaded] = useState('false');

	useEffect(() => {
		if (!src) {
			return undefined;
		}

		setLoaded('loading');

		let active = true;
		const image = new Image();
		image.src = src;
		image.onload = () => {
			if (!active) {
				return;
			}
			setLoaded('loaded');
		};
		image.onerror = () => {
			if (!active) {
				return;
			}
			setLoaded('error');
		};

		return () => {
			active = false;
		};
	}, [src]);

	return loaded;
}
