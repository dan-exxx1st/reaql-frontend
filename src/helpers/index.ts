import { parse } from 'querystring';

export function getDialogIdFromSearch(search: string) {
    const params = parse(search);

    const dialogId = Object.entries(params)
        .map(([key, value]) => key.indexOf('dialog') !== -1 && value)
        .toString();

    return dialogId;
}
