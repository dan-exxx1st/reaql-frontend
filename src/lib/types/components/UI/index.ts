import { MouseEventHandler } from 'react';
import Theme from 'helpers/styled';
import { ColorType, IconName } from 'lib/types';
import { TypographyVariantType, TypographyComponentType } from 'lib/types/ui';

export interface IAvatarProps {
    src?: HTMLImageElement['src'];
    alt?: string;
    className?: string;
}

export interface IButtonProps {
    className?: string;
    theme: typeof Theme;
    onClick?: MouseEventHandler;
    variant: 'filled' | 'outlined';
    color: ColorType;
}

export interface ICheckMarkProps {
    isChecked: boolean;
    isDouble: boolean;
}

export interface ICircleProps {
    color: ColorType;
    iconName?: IconName;
    text?: string | number;
    width?: string;
    height?: string;
    cursor?: string;
    className?: string;
    onClick?: () => void;
}

export interface IIconProps {
    iconName: IconName;
    className?: string;
}

export interface ISkeletonLoaderProps {
    variant?: 'avatar' | 'block';
    width?: string;
    height?: string;
    className?: string;
}

export interface ITextFieldProps {
    onChange?: (value: string) => void;
    value?: string;
    className?: string;
    fieldSize: 'small' | 'medium' | 'large';
    color: ColorType;
    icon?: IconName;
    width?: string;
    height?: string;
    wrapperWidth?: string;
    placeholder?: string;
    type?: string;
}

export interface ITypographyProps {
    className?: string;
    variant: TypographyVariantType;
    component?: TypographyComponentType;
    color?: string;
    theme: typeof Theme;
}

export type { IWrapperProps } from './WrapperTypes';

export interface ICheckBoxProps {
    onClick?: () => void;
    text?: string;
}
