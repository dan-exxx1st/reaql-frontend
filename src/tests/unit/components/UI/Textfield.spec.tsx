import React from 'react';
import { TextField } from 'components/UI';
import { mountWithTheme } from 'tests/helpers/withTheme';

describe('<TextField />', () => {
	describe('Snapshots', () => {
		it('Primary text field with large size to match snapshot', () => {
			const wrapper = mountWithTheme(
				<TextField
					color="primary"
					fieldSize="large"
					value="TextField snapshot"
				/>
			);
			expect(wrapper).toMatchSnapshot();
		});
		it('Primary text field with medium size to match snapshot', () => {
			const wrapper = mountWithTheme(
				<TextField
					color="primary"
					fieldSize="medium"
					value="TextField snapshot"
				/>
			);
			expect(wrapper).toMatchSnapshot();
		});
		it('Primary text field with small size to match snapshot', () => {
			const wrapper = mountWithTheme(
				<TextField
					color="primary"
					fieldSize="small"
					value="TextField snapshot"
				/>
			);
			expect(wrapper).toMatchSnapshot();
		});
		it('Secondary text field with large size to match snapshot', () => {
			const wrapper = mountWithTheme(
				<TextField
					color="secondary"
					fieldSize="large"
					value="TextField snapshot"
				/>
			);
			expect(wrapper).toMatchSnapshot();
		});
		it('Secondary text field with medium size to match snapshot', () => {
			const wrapper = mountWithTheme(
				<TextField
					color="secondary"
					fieldSize="medium"
					value="TextField snapshot"
				/>
			);
			expect(wrapper).toMatchSnapshot();
		});
		it('Secondary text field with small size to match snapshot', () => {
			const wrapper = mountWithTheme(
				<TextField
					color="secondary"
					fieldSize="small"
					value="TextField snapshot"
				/>
			);
			expect(wrapper).toMatchSnapshot();
		});
	});

	it('Should change value in text field', () => {
		const changeValue = jest.fn();
		const wrapper = mountWithTheme(
			<TextField
				color="primary"
				fieldSize="small"
				onChange={changeValue}
			/>
		);
		wrapper.simulate('change');
		expect(changeValue).toBeCalled();
	});
});
