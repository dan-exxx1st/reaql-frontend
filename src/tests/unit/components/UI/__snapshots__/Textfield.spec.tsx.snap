// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> Snapshots Primary text field with custom width and height 1`] = `
.c0 {
  position: static;
}

.c1 {
  word-break: break-all;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-family: 'Roboto',sans-serif;
  padding: 10px;
  border: 1px solid #E6E6E6;
  border-radius: 5px;
  outline: none;
  color: #555555;
  cursor: text;
  font-size: 12px;
  font-weight: normal;
  line-height: 14px;
  -webkit-letter-spacing: 0.4px;
  -moz-letter-spacing: 0.4px;
  -ms-letter-spacing: 0.4px;
  letter-spacing: 0.4px;
  width: 25px;
  height: 25px;
  padding-right: 0;
}

.c1 ~ img {
  width: 20px;
  height: 20px;
  position: absolute;
  top: 50%;
  right: 10px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

<div
  className="c0"
>
  <input
    className="c1"
    color="secondary"
    height="25px"
    onChange={[Function]}
    width="25px"
  />
</div>
`;

exports[`<TextField /> Snapshots Primary text field with icon 1`] = `
.c2 {
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: 0;
  margin: 0;
}

.c0 {
  position: relative;
}

.c1 {
  word-break: break-all;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-family: 'Roboto',sans-serif;
  padding: 10px;
  border: 1px solid #E6E6E6;
  border-radius: 5px;
  outline: none;
  color: #555555;
  cursor: text;
  font-size: 12px;
  font-weight: normal;
  line-height: 14px;
  -webkit-letter-spacing: 0.4px;
  -moz-letter-spacing: 0.4px;
  -ms-letter-spacing: 0.4px;
  letter-spacing: 0.4px;
  width: 170px;
  height: auto;
  padding-right: 40px;
}

.c1 ~ img {
  width: 20px;
  height: 20px;
  position: absolute;
  top: 50%;
  right: 10px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

<div
  className="c0"
>
  <input
    className="c1"
    color="secondary"
    onChange={[Function]}
  />
  <img
    alt="Icon component"
    className="c2"
    src="SearchIcon.svg"
  />
</div>
`;

exports[`<TextField /> Snapshots Primary text field with large size  1`] = `
.c0 {
  position: static;
}

.c1 {
  word-break: break-all;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-family: 'Roboto',sans-serif;
  padding: 10px;
  border: 1px solid #00ADFF;
  border-radius: 5px;
  outline: none;
  color: #252525;
  cursor: text;
  font-size: 16px;
  font-weight: normal;
  line-height: 19px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  width: 220px;
  height: auto;
  padding-right: 0;
}

.c1 ~ img {
  width: 20px;
  height: 20px;
  position: absolute;
  top: 50%;
  right: 10px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

<div
  className="c0"
>
  <input
    className="c1"
    color="primary"
    onChange={[Function]}
    value="TextField snapshot"
  />
</div>
`;

exports[`<TextField /> Snapshots Primary text field with medium size 1`] = `
.c0 {
  position: static;
}

.c1 {
  word-break: break-all;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-family: 'Roboto',sans-serif;
  padding: 10px;
  border: 1px solid #00ADFF;
  border-radius: 5px;
  outline: none;
  color: #252525;
  cursor: text;
  font-size: 14px;
  font-weight: normal;
  line-height: 16px;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  width: 190px;
  height: auto;
  padding-right: 0;
}

.c1 ~ img {
  width: 20px;
  height: 20px;
  position: absolute;
  top: 50%;
  right: 10px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

<div
  className="c0"
>
  <input
    className="c1"
    color="primary"
    onChange={[Function]}
    value="TextField snapshot"
  />
</div>
`;

exports[`<TextField /> Snapshots Primary text field with small size  1`] = `
.c0 {
  position: static;
}

.c1 {
  word-break: break-all;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-family: 'Roboto',sans-serif;
  padding: 10px;
  border: 1px solid #00ADFF;
  border-radius: 5px;
  outline: none;
  color: #252525;
  cursor: text;
  font-size: 12px;
  font-weight: normal;
  line-height: 14px;
  -webkit-letter-spacing: 0.4px;
  -moz-letter-spacing: 0.4px;
  -ms-letter-spacing: 0.4px;
  letter-spacing: 0.4px;
  width: 170px;
  height: auto;
  padding-right: 0;
}

.c1 ~ img {
  width: 20px;
  height: 20px;
  position: absolute;
  top: 50%;
  right: 10px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

<div
  className="c0"
>
  <input
    className="c1"
    color="primary"
    onChange={[Function]}
    value="TextField snapshot"
  />
</div>
`;

exports[`<TextField /> Snapshots Secondary text field with large size 1`] = `
.c0 {
  position: static;
}

.c1 {
  word-break: break-all;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-family: 'Roboto',sans-serif;
  padding: 10px;
  border: 1px solid #E6E6E6;
  border-radius: 5px;
  outline: none;
  color: #555555;
  cursor: text;
  font-size: 16px;
  font-weight: normal;
  line-height: 19px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  width: 220px;
  height: auto;
  padding-right: 0;
}

.c1 ~ img {
  width: 20px;
  height: 20px;
  position: absolute;
  top: 50%;
  right: 10px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

<div
  className="c0"
>
  <input
    className="c1"
    color="secondary"
    onChange={[Function]}
    value="TextField snapshot"
  />
</div>
`;

exports[`<TextField /> Snapshots Secondary text field with medium size 1`] = `
.c0 {
  position: static;
}

.c1 {
  word-break: break-all;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-family: 'Roboto',sans-serif;
  padding: 10px;
  border: 1px solid #E6E6E6;
  border-radius: 5px;
  outline: none;
  color: #555555;
  cursor: text;
  font-size: 14px;
  font-weight: normal;
  line-height: 16px;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  width: 190px;
  height: auto;
  padding-right: 0;
}

.c1 ~ img {
  width: 20px;
  height: 20px;
  position: absolute;
  top: 50%;
  right: 10px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

<div
  className="c0"
>
  <input
    className="c1"
    color="secondary"
    onChange={[Function]}
    value="TextField snapshot"
  />
</div>
`;

exports[`<TextField /> Snapshots Secondary text field with small size 1`] = `
.c0 {
  position: static;
}

.c1 {
  word-break: break-all;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-family: 'Roboto',sans-serif;
  padding: 10px;
  border: 1px solid #E6E6E6;
  border-radius: 5px;
  outline: none;
  color: #555555;
  cursor: text;
  font-size: 12px;
  font-weight: normal;
  line-height: 14px;
  -webkit-letter-spacing: 0.4px;
  -moz-letter-spacing: 0.4px;
  -ms-letter-spacing: 0.4px;
  letter-spacing: 0.4px;
  width: 170px;
  height: auto;
  padding-right: 0;
}

.c1 ~ img {
  width: 20px;
  height: 20px;
  position: absolute;
  top: 50%;
  right: 10px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

<div
  className="c0"
>
  <input
    className="c1"
    color="secondary"
    onChange={[Function]}
    value="TextField snapshot"
  />
</div>
`;
